src/openrct2/game.c

```diff
@@ -1044,8 +1044,9 @@ void game_autosave()
 	platform_get_time_local(&currentTime);
 
 	utf8 timeName[34];
-	snprintf(timeName, 34, "autosave_%d-%02d-%02d_%02d-%02d-%02d%s",
-		currentDate.year, currentDate.month, currentDate.day, currentTime.hour, currentTime.minute,currentTime.second,
+	snprintf(timeName, 34, "autosave_%04u-%02u-%02u_%02u-%02u-%02u%s",
+		currentDate.year % 10000, currentDate.month % 100, currentDate.day % 100,
+		currentTime.hour % 100, currentTime.minute % 100, currentTime.second % 100,
```

Comment:
I think you should also change the day, month, day_of_week, hour, minute, and second fields in those structs to uint8.